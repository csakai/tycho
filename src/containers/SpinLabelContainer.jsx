import React from 'react';
import {connect} from 'react-redux';
import SpinLabel from '../components/SpinLabel';
import ReduxService from '../services/ReduxService';

export class SpinLabelContainer extends React.Component {

  /**
   * Calculate visibility of label based on props.
   *
   * @returns {Boolean} visibility of label
   */
  isVisible = () => {
    const {isComplete, isAutoOrbitEnabled} = this.props;
    return isComplete && isAutoOrbitEnabled;
  }

  render() {
    return (
      <SpinLabel
        show={this.isVisible()}
        count={4}
      />
    );
  }
}

export default connect(
  ReduxService.mapStateToProps(
    'tour.isComplete',
    'tour.isAutoOrbitEnabled'
  ),
  null  
)(SpinLabelContainer);
