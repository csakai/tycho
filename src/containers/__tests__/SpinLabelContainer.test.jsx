import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import {SpinLabelContainer} from '../SpinLabelContainer';

describe('Spin Label Container', () => {
  let component, spinLabelContainer;

  beforeEach(() => {
    component = shallow(<SpinLabelContainer />);
    spinLabelContainer = component.instance();
  });

  describe('isVisible()', () => {
    it('should return true if both isAutoOrbitEnabled and isComplete props are true', () => {
      spinLabelContainer.props = {
        isAutoOrbitEnabled: true,
        isComplete: true
      };
      expect(spinLabelContainer.isVisible()).toEqual(true);
    });

    it('should return true if both isAutoOrbitEnabled is false', () => {
      spinLabelContainer.props = {
        isAutoOrbitEnabled: false,
        isComplete: true
      };
      expect(spinLabelContainer.isVisible()).toEqual(false);
    });

    it('should return true if both isComplete is false', () => {
      spinLabelContainer.props = {
        isAutoOrbitEnabled: true,
        isComplete: false
      };
      expect(spinLabelContainer.isVisible()).toEqual(false);
    });
  });

  describe('render()', () => {
    it('should render the spin label container successfully', () => {
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
