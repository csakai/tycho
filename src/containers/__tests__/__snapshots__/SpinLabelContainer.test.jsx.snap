// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spin Label Container render() should render the spin label container successfully 1`] = `
<SpinLabel
  count={4}
/>
`;
